tests/testthat/test-connect.R

```diff
@@ -39,6 +39,21 @@ test_that("Handling error responses", {
   expect_error(con$raise_error(resp), "Bad Request")
 })
 
+test_that("Handling deprecation warnings", {
+  resp <- fake_response("https://connect.example/__api__/", headers = list(
+    `X-Deprecated-Endpoint` = "/v1"
+  ))
+  expect_warning(
```

Comment:
Do we already confirm that a warning is not generated in the absence of the header?

Reply:
If every request raised this warning, the test output would be a mess, and it's not. But I can add an explicit test here.